kotlin/perform-valve-isolation-trace/src/main/java/com/esri/arcgisruntime/sample/performvalveisolationtrace/MainActivity.kt

```diff
@@ -0,0 +1,295 @@
+import com.esri.arcgisruntime.symbology.SimpleRenderer
+import com.esri.arcgisruntime.utilitynetworks.UtilityCategory
+import com.esri.arcgisruntime.utilitynetworks.UtilityCategoryComparison
+import com.esri.arcgisruntime.utilitynetworks.UtilityCategoryComparisonOperator
+import com.esri.arcgisruntime.utilitynetworks.UtilityElement
+import com.esri.arcgisruntime.utilitynetworks.UtilityElementTraceResult
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetwork
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceConfiguration
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceFilter
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceParameters
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceType
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.spinner_text_item.view.*
+import java.util.UUID
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG = MainActivity::class.java.simpleName
+
+  // create a graphics overlay for the starting location and add it to the map view
+  private val startingLocationGraphicsOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+
+  // create and apply renderers for the starting point graphics overlay
+  private val startingPointSymbol: SimpleMarkerSymbol by lazy {
+    SimpleMarkerSymbol(
+      SimpleMarkerSymbol.Style.CROSS,
+      Color.GREEN,
+      25f
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // load the utility network data from the feature service and create feature layers
+    val distributionLineFeatureTable =
+      ServiceFeatureTable(getString(R.string.distribution_line_url))
+    val distributionLineLayer = FeatureLayer(distributionLineFeatureTable)
+    val deviceFeatureTable =
+      ServiceFeatureTable(getString(R.string.device_url))
+    val deviceLayer = FeatureLayer(deviceFeatureTable)
+
+    // create a map with the utility network distribution line and device layers
+    val map = ArcGISMap(Basemap.createStreetsNightVector()).apply {
+      // add the feature layers to the map
+      operationalLayers.addAll(listOf(distributionLineLayer, deviceLayer))
+      // create and load the utility network
+      addDoneLoadingListener {
+        createUtilityNetwork()
+      }
+    }
+
+    mapView.apply {
+      this.map = map
+      // set the starting location graphic overlay's renderer and add it to the map view
+      startingLocationGraphicsOverlay.renderer = SimpleRenderer(startingPointSymbol)
+      graphicsOverlays.add(startingLocationGraphicsOverlay)
+
+      // make sure the fab doesn't obscure the attribution bar
+      addAttributionViewLayoutChangeListener { _, _, _, _, bottom, _, _, _, oldBottom ->
+        val layoutParams = fab.layoutParams as CoordinatorLayout.LayoutParams
+        layoutParams.bottomMargin += bottom - oldBottom
```

Comment:
this worked fine in the sample viewer...don't know why it acted strangely in the attribution view change sample...